File: UpDownHiking/solve/ruby/UpDownHikingSolver.rb
#! /usr/bin/env ruby

require_relative "UpDownHiking"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  N = reader.next(TInt)
  reader.next()
  A = reader.next(TInt)
  reader.next()
  B = reader.next(TInt)

  result = UpDownHiking.new().maxHeight(N, A, B)
  IO.write(ARGV[1], Writer.new.next(result, TInt).to_s)
end

init
main

File: UpDownHiking/solve/cpp/UpDownHikingSolver.cpp
#include <fstream>
#include <topcoder>
#include "UpDownHiking.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int N; tc::read(ifs, N); tc::next(ifs);
    int A; tc::read(ifs, A); tc::next(ifs);
    int B; tc::read(ifs, B);
    ifs.close();

    std::ofstream ofs(argv[2]);
    UpDownHiking solver;
    tc::show(ofs, solver.maxHeight(N, A, B));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}
